java/opendcs/src/main/java/decodes/util/ExportTimeSeries.java

```diff
@@ -151,60 +158,61 @@ protected void runApp()
 		until = convert2Date(s, true);
 
 		ArrayList<CTimeSeries> ctss = new ArrayList<CTimeSeries>();
-		TimeSeriesDAI tsDAO = theDb.makeTimeSeriesDAO();
-		for(int n = tsidArg.NumberOfValues(), i=0; i<n; i++)
+		try (TimeSeriesDAI tsDAO = theDb.makeTimeSeriesDAO())
 		{
-			String outTS = tsidArg.getValue(i);
-			if (outTS.equalsIgnoreCase("all"))
-			{
-				ArrayList<TimeSeriesIdentifier> tsids = tsDAO.listTimeSeries();
-				for(TimeSeriesIdentifier tsid : tsids)
-				{
-					CTimeSeries ts = theDb.makeTimeSeries(tsid);
-					int nvalues = tsDAO.fillTimeSeries(ts, since, until);
-					Logger.instance().info("Read " + nvalues + " values for time series " 
-						+ tsid.getUniqueString());
-					ctss.add(ts);
-				}
-				break; // No need to continue, we have all time series now.
-			}
-			else if (TextUtil.startsWithIgnoreCase(outTS, "group:"))
+			for(int n = tsidArg.NumberOfValues(), i=0; i<n; i++)
 			{
-				String groupName = outTS.substring(6);
-				TsGroupDAI groupDAO = theDb.makeTsGroupDAO();
-				TsGroup grp = groupDAO.getTsGroupByName(groupName);
-				groupDAO.close();
-				if (grp == null)
+				String outTS = tsidArg.getValue(i);
+				if (outTS.equalsIgnoreCase("all"))
 				{
-					Logger.instance().warning("No such time series group: " + groupName + " -- skipped.");
-					continue;
-				}
-				ArrayList<TimeSeriesIdentifier> tsids = theDb.expandTsGroup(grp);
-				for(TimeSeriesIdentifier tsid : tsids)
-				{
-					CTimeSeries ts = theDb.makeTimeSeries(tsid);
-					int nvalues = tsDAO.fillTimeSeries(ts, since, until);
-					Logger.instance().info("Read " + nvalues + " values for time series " 
-						+ tsid.getUniqueString());
-					ctss.add(ts);
+					ArrayList<TimeSeriesIdentifier> tsids = tsDAO.listTimeSeries();
+					for(TimeSeriesIdentifier tsid : tsids)
+					{
+						CTimeSeries ts = theDb.makeTimeSeries(tsid);
+						int nvalues = tsDAO.fillTimeSeries(ts, since, until);
+						Logger.instance().info("Read " + nvalues + " values for time series "
+							+ tsid.getUniqueString());
+						ctss.add(ts);
+					}
+					break; // No need to continue, we have all time series now.
 				}
-			}
-			else // Should be a time series ID
-			{
-				try
+				else if (TextUtil.startsWithIgnoreCase(outTS, "group:"))
 				{
-					CTimeSeries ts = theDb.makeTimeSeries(outTS);
-					int nvalues = tsDAO.fillTimeSeries(ts, since, until);
-					Logger.instance().info("Read " + nvalues + " values for time series " + outTS);
-					ctss.add(ts);
+					String groupName = outTS.substring(6);
```

Comment:
Suggestion: magic number 6 needs a comment to  explain.

Reply:
hmm, yeah. length of the word "group:" should define group as a constant and use length.